src/common/io_file.cpp

```diff
@@ -216,6 +218,26 @@ void IOFile::Close() {
 #endif
 }
 
+void IOFile::Unlink() {
+    if (!IsOpen()) {
+        return;
+    }
+
+    // Mark the file for deletion
+    // TODO: Also remove the file path?
+#if _WIN64
+    FILE_DISPOSITION_INFORMATION disposition;
+    IO_STATUS_BLOCK iosb;
+
+    const int fd = fileno(file);
+    HANDLE hfile = reinterpret_cast<HANDLE>(_get_osfhandle(fd));
+
+    disposition.DeleteFile = TRUE;
+    NtSetInformationFile(hfile, &iosb, &disposition, sizeof(disposition),
+                         FileDispositionInformation);
+#endif
```

Comment:
Add a #else here with UNREACHABLE() so when this is called on linux it won't silently fail